src/main/scala/konstructs/shard.scala

```diff
@@ -69,19 +71,34 @@ case class ChunkData(version: Int, data: Array[Byte]) {
 }
 
 object ChunkData {
+  val InitialRevision = 1
   val Size = Db.ChunkSize * Db.ChunkSize * Db.ChunkSize * Db.BlockSize
 
-  def apply(blocks: Array[Byte], buffer: Array[Byte]): ChunkData = {
+  def writeRevision(data: Array[Byte], revision: Int, offset: Int) {
+    data(offset + 0) = (revision & 0xFF).toByte
+    data(offset + 1) = ((revision >> 8) & 0xFF).toByte
+    data(offset + 2) = ((revision >> 16) & 0xFF).toByte
+    data(offset + 3) = ((revision >> 24) & 0x7F).toByte
```

Comment:
On second thought, maybe it is simpler just to keep the shift/and operators to keep the logic the same on both server and client. Do not really care, was just a thought.